packages/puppeteer-core/src/api/ElementHandle.ts

```diff
@@ -886,7 +886,25 @@ export class ElementHandle<
    * or `null` if the element is not visible.
    */
   async boundingBox(): Promise<BoundingBox | null> {
-    throw new Error('Not implemented');
+    if (this.frame.parentFrame()) {
+      throw new Error(
+        'Elements within nested iframes are currently not supported.'
+      );
+    }
+    // TODO(jrandolf): Use sandboxes when available.
+    const box = await this.evaluate(element => {
+      if ('offsetParent' in element && element.offsetParent === null) {
```

Comment:
Can you add the link to MDN explaining this part. Else it feels random check.

Reply:
I've removed the check for now. I think this needs to be reevaluated more thoroughly.